tests/apps/search/test_query_courses.py

```diff
@@ -272,18 +120,288 @@ def get_expected_courses(courses_definition, course_run_ids):
+        - create relevant objects that will be linked with courses,
         - generate a set of courses randomly associated to our "interesting" course runs,
-        - prepare the Elasticsearch index,
-        - execute the query.
+        - prepare the Elasticsearch index.
         """
+        filter_pages = self.create_filter_pages()
+
+        top_subjects = [
+            # 00010001 througL 00010004
+            CategoryFactory(
+                page_parent=filter_pages["subjects"].extended_object,
+                page_title=f"Subject #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 4)
+        ]
+
+        subject_0_children = [
+            # 000100010001 and 000100010002
+            CategoryFactory(
+                page_parent=top_subjects[0].extended_object,
+                page_title=f"Subject #0 Child #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 2)
+        ]
+
+        subject_1_children = [
+            # 000100020001 and 000100020002
+            CategoryFactory(
+                page_parent=top_subjects[1].extended_object,
+                page_title=f"Subject #1 Child #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 2)
+        ]
+
+        levels = [
+            # 00020001 through 00020003
+            CategoryFactory(
+                page_parent=filter_pages["levels"].extended_object,
+                page_title=f"Level #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 3)
+        ]
+
+        top_organizations = [
+            # 00030001 through 00030004
+            OrganizationFactory(
+                page_parent=filter_pages["organizations"].extended_object,
+                page_title=f"Organization #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 4)
+        ]
+
+        organization_0_children = [
+            # 000300010001 and 000300010002
+            OrganizationFactory(
+                page_parent=top_organizations[0].extended_object,
+                page_title=f"Organization #0 Child #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 2)
+        ]
+
+        organization_1_children = [
+            # 000300020001 and 000300020002
+            OrganizationFactory(
+                page_parent=top_organizations[1].extended_object,
+                page_title=f"Organization #1 Child #{i}",
+                should_publish=True,
+            )
+            for i in range(0, 2)
+        ]
+
+        self.persons = [
+            PersonFactory(
+                page_title="Mikhaïl Boulgakov",
+                should_publish=True,
+            )
+        ]
+
+        # NB: Indexed human names do not match the actual titles for the pages.
+        # This makes it simpler to run full-text searches and check their results.
+        courses = [
```

Comment:
I didn't get why this is hardcoded and not generated via the indexer's method. Since we are testing the queries and not the indexer's document formatting it would be more robust and simpler?

Reply:
I tried to make the fewest changes I could get away with, and decided not to touch this.

I also think it makes the tests less brittle: if we're testing search queries, we have to hardcode every bit of searchable data, otherwise the tests could have different results between runs.

Reply:
ok ok